### PR TITLE
Return 409 response when trying to register existing username

### PR DESCRIPTION
**Pull request by joaxcar**
The PR solves the issue of missing information when a user tries to register an existing account.

The frontend code is somewhat crude. It just checks for 409 and in the registration page assumes this is a username conflict. This results in a message "Username exists" and the username field is emptied.

The registration fields are now only reset'ed if there is a succes.

Also cleaned up some test code and some javadocs.

This code does not solve the possibility to have duplicates in the database. It only stops new registrations to enter the database if one exist

### CODE REVIEW

File: frontend/src/containers/RegistrationContainer/RegistrationContainer.js [hunk 1]
```diff
@@ -45,17 +45,23 @@ function RegistrationContainer() {
...
 	function handleFormSubmit(registrationDetails, event) {
 		event.preventDefault();
-		event.target.reset();
-
 		const formattedRegistrationDetails = formatDetails(registrationDetails);
 
 		postToAPI(endpoints.ACCOUNTS, formattedRegistrationDetails)
-			.then((response) =>
+			.then((response) => {
+				console.log(response)
```
Comment (joaxcar) [inline frontend/src/containers/RegistrationContainer/RegistrationContainer.js]: Remove

File: frontend/src/containers/RegistrationContainer/RegistrationContainer.js [hunk 2]
```diff
@@ -45,17 +45,23 @@ function RegistrationContainer() {
...
-			.catch((error) =>
-				showAlert(alertTypes.ERROR, "Something went wrong")
-			);
+			})
+			.catch((error) =>{
+				if(error.message === "Conflict") {
+					document.getElementsByName('username')[0].value = "";
+					showAlert(alertTypes.ERROR, "Existing username")
+				} else {
+					console.log(error)
```
Comment (joaxcar) [inline frontend/src/containers/RegistrationContainer/RegistrationContainer.js]: remove

File: frontend/src/containers/RegistrationContainer/RegistrationContainer.js [hunk 3]
```diff
@@ -45,17 +45,23 @@ function RegistrationContainer() {
...
 				showAlert(alertTypes.SUCCESS, "Registration was successful")
-			)
-			.catch((error) =>
-				showAlert(alertTypes.ERROR, "Something went wrong")
-			);
+			})
+			.catch((error) =>{
+				if(error.message === "Conflict") {
+					document.getElementsByName('username')[0].value = "";
+					showAlert(alertTypes.ERROR, "Existing username")
```
Comment (RW90) [inline frontend/src/containers/RegistrationContainer/RegistrationContainer.js]: Change error message to "User already exists"

**Review by RW90 — APPROVED** [review]
Good work! Nice refarctoring of the tests.

**Review by nedvin — APPROVED** [review]
Looks good!

**Pull request merged**